hydra-node/test/Hydra/Chain/Direct/TxSpec.hs

```diff
@@ -141,6 +187,215 @@ spec =
+                , let blueprintMetadata = fromSMaybe mempty $ getAuxMetadata <$> blueprintTx ^. auxDataTxL
+                      commitMetadata = fromSMaybe mempty $ getAuxMetadata <$> tx ^. auxDataTxL
+                   in blueprintMetadata `Map.isSubmapOf` commitMetadata
+                        & counterexample ("blueprint metadata: " <> show blueprintMetadata)
+                        & counterexample ("commit metadata: " <> show commitMetadata)
+                , let blueprintValidity = blueprintBody ^. vldtTxBodyL
+                      commitValidity = commitTxBody ^. vldtTxBodyL
+                   in blueprintValidity === commitValidity
+                        & counterexample ("blueprint validity: " <> show blueprintValidity)
+                        & counterexample ("commit validity: " <> show commitValidity)
+                , let blueprintInputs = blueprintBody ^. inputsTxBodyL
+                      commitInputs = commitTxBody ^. inputsTxBodyL
+                   in property (blueprintInputs `Set.isSubsetOf` commitInputs)
+                        & counterexample ("blueprint inputs: " <> show blueprintInputs)
+                        & counterexample ("commit inputs: " <> show commitInputs)
+                , let blueprintOutputs = toList $ blueprintBody ^. outputsTxBodyL
+                      commitOutputs = toList $ commitTxBody ^. outputsTxBodyL
+                   in property
+                        ( all
+                            (`notElem` blueprintOutputs)
+                            commitOutputs
+                        )
+                        & counterexample ("blueprint outputs: " <> show blueprintOutputs)
+                        & counterexample ("commit outputs: " <> show commitOutputs)
+                , let blueprintSigs = blueprintBody ^. reqSignerHashesTxBodyL
+                      commitSigs = commitTxBody ^. reqSignerHashesTxBodyL
+                   in property (blueprintSigs `Set.isSubsetOf` commitSigs)
+                        & counterexample ("blueprint signatures: " <> show blueprintSigs)
+                        & counterexample ("commit signatures: " <> show commitSigs)
+                , let blueprintRefInputs = blueprintBody ^. referenceInputsTxBodyL
+                      commitRefInputs = commitTxBody ^. referenceInputsTxBodyL
+                   in property (blueprintRefInputs `Set.isSubsetOf` commitRefInputs)
+                        & counterexample ("blueprint reference inputs: " <> show blueprintRefInputs)
+                        & counterexample ("commit reference inputs: " <> show commitRefInputs)
+                ]
+                & cover 1 (spendsFromScript (lookupUTxO, fromLedgerTx blueprintTx)) "blueprint spends script UTxO"
+                & cover 1 (spendsFromPubKey (lookupUTxO, fromLedgerTx blueprintTx)) "blueprint spends pub key UTxO"
+                & cover 1 (not . null $ blueprintTx ^. bodyTxL . referenceInputsTxBodyL) "blueprint has reference input"
```

Comment:
As we do test the generator producing `lookupUTxO` and `blueprintTx`, we do not need to check coverage on this test here anymore. Hence the `spendsFromPubKey` and `spendsFromScript` could also stay local to that test.